From 4e6a7ad1d66d170f06f503c78b8b137aeef8a8e7 Mon Sep 17 00:00:00 2001
From: Jerry Jelinek <jerry.jelinek@joyent.com>
Date: Wed, 1 Mar 2017 15:54:47 +0000
Subject: [PATCH] OS-5990 fix for OS-5987 causes worker thread to mishandle
 some signals Reviewed by: Patrick Mooney <patrick.mooney@joyent.com> Approved
 by: Patrick Mooney <patrick.mooney@joyent.com>

---
 usr/src/uts/common/brand/lx/syscall/lx_aio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_aio.c b/usr/src/uts/common/brand/lx/syscall/lx_aio.c
index 57a6eae103..027c89ff99 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_aio.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_aio.c
@@ -736,7 +736,7 @@ lx_io_setup(uint_t nr_events, void *ctxp)
 		mutex_enter(&curproc->p_lock);
 		t->t_proc_flag = (t->t_proc_flag & ~TP_HOLDLWP) | TP_KTHREAD;
 		lwptolxlwp(l)->br_lwp_flags |= BR_AIO_LWP;
-		curthread->t_hold = hold_set;
+		t->t_hold = hold_set;
 		lwp_create_done(t);
 		mutex_exit(&curproc->p_lock);
 	}
-- 
2.21.0

